examples/lib/intermediateHtmlStepUi.js

```diff
@@ -104,61 +95,43 @@ function IntermediateHtmlStepUi(_sequencer, step, options) {
       });
     }
 
-    $step('.insertDiv .close-insert-box').off('click').on('click', function(){toggleDiv(function(){});});
+    $step('.insertDiv .close-insert-box').off('click').on('click', function(){
+      toggleDiv($step);
+      $step('.insertDiv').removeClass('insertDiv');
+    });
     
     var insertStepSelect = $step('.insert-step-select');
     insertStepSelect.html('');
     // Add modules to the insertStep dropdown
     for (var m in modulesInfo) {
-      if (modulesInfo[m] !== undefined)
+      if (modulesInfo[m] && modulesInfo[m].name)
         insertStepSelect.append(
           '<option value="' + m + '">' + modulesInfo[m].name + '</option>'
         );
     }
     insertStepSelect.selectize({
       sortField: 'text'
     });
-    $step('.inserDiv .add-step-btn').prop('disabled', true);
-    
-    insertStepSelect.append('<option value="" disabled selected>Select a Module</option>');
-    $step('.insertDiv .radio-group .radio').on('click', function () {
-      $(this).parent().find('.radio').removeClass('selected');
-      $(this).addClass('selected');
-      newStep = $(this).attr('data-value');
-      $step('.insert-step-select').val(newStep);
-      selectNewStepUi($step);
-      insert(id, $step);
-      $(this).removeClass('selected');
+    $('.insertDiv .radio-group .radio').on('click', function () {
+      var newStepName = $(this).attr('data-value');
+      id = $($step('.insertDiv').parents()[3]).prevAll().length;
+      insert(id, $step, newStepName);
     });
-    insertStepSelect.on('change', () => {selectNewStepUi($step);});
-    $step('.insertDiv .add-step-btn').on('click', function () { insert(id, $step); });
+    $step('.insertDiv .add-step-btn').on('click', function () {
+      var newStepName = insertStepSelect.val();
+      id = $($step('.insertDiv').parents()[3]).prevAll().length;
+      insert(id, $step, newStepName); });
   };
 
-  function insert(id, $step) {
-
-    options = options || {};
-    var insertStepSelect = $step('.insert-step-select');
-    if (insertStepSelect.val() == 'none') return;
-
-    var newStepName = insertStepSelect.val();
+  function insert(id, $step, newStepName) {
```

Comment:
This is looking good -- do you think we could make the name here more specific? Like, insert what? Insert step, right? 

Thank you!!!

Reply:
@jywarren was thinking the same but insertstep was already defined in line 77,so left as it is